vendor/engines/bulk_email/app/decorators/bulk_email/job_decorator.rb

```diff
@@ -0,0 +1,42 @@
+module BulkEmail
+
+  class JobDecorator < SimpleDelegator
+
+    def sender
+      user.email
+    end
+
+    def recipient_list
+      recipients.join(", ")
+    end
+
+    def products
+      Product.find(selected_product_ids).map(&:name).join(", ")
```

Comment:
I created an email where I selected no products (so not through instrument not available). This is causing the details view to 404 because it's trying to do `SELECT  `products`.* FROM `products` WHERE `products`.`id` = 0 LIMIT 1`